FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+import { Link } from 'react-router-dom';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { useState, useMemo } from 'react';
+import { useSWRConfig } from 'swr';
+import useSWRMutation from 'swr/mutation';
+import styles from './CompanyCard.module.css';
+import { useUser } from '../../hooks';
+import PropTypes from 'prop-types';
+import { Tooltip, Badge } from 'antd';
+
+export default function CompanyCard({ data, isAuthorized }) {
+  CompanyCard.propTypes = {
+    data: PropTypes.object,
+    isAuthorized: PropTypes.bool,
+  };
+  const [isSaved, setIsSaved] = useState(data && data.is_saved);
+  const { mutate } = useSWRConfig();
+  const profile = useMemo(() => {
+    return {
+      id: data?.id,
+      personId: data?.person,
+      name: data?.name,
+      region: data?.region,
+      categories: data?.categories,
+      activities: data?.activities,
+      logo: data?.logo_image,
+      banner: data?.banner_image,
+      founded: data?.founded,
+      isSaved: data?.is_saved,
+    };
+  }, [data]);
+
+  const { user } = useUser();
+  const ownProfile = user && user.id === profile.personId;
+
+  async function sendRequest(url, { arg: data }) {
+    const authToken = localStorage.getItem('Token');
```

Comment:
FYI: I asked Roman to return token from `useProvideAuth` when `isAuth` `true`. In the future it's better to use token from the hook.

Reply:
It's in the Lilia's PR. She worked on it. She's waiting for your review. We'll be able change it on all pages after her merge